src/providers/GaugeProvider.tsx

```diff
@@ -0,0 +1,158 @@
+import {
+  ChainId,
+  GAUGE_CONTROLLER_ADDRESSES,
+  HELPER_CONTRACT_ADDRESSES,
+} from "../constants"
+import React, { ReactElement, useEffect, useState } from "react"
+import {
+  createMultiCallContract,
+  enumerate,
+  getMulticallProvider,
+} from "../utils"
+import {
+  useGaugeControllerContract,
+  useHelperContract,
+} from "../hooks/useContract"
+import { BigNumber } from "ethers"
+import GAUGE_CONTROLLER_ABI from "../constants/abis/gaugeController.json"
+import { GaugeController } from "../../types/ethers-contracts/GaugeController"
+import HELPER_CONTRACT_ABI from "../constants/abis/helperContract.json"
+import { HelperContract } from "../../types/ethers-contracts/HelperContract"
+import { Web3Provider } from "@ethersproject/providers"
+import { useActiveWeb3React } from "../hooks"
+
+export type Gauge = {
+  address: string
+  gaugeWeight: BigNumber
+  poolAddress: string
+  gaugeRelativeWeight: BigNumber
+  workingSupply?: BigNumber
```

Comment:
can we make this non-optional?

Reply:
I can, but we're going to have to initialize it with an empty value.